pages/index.js

```diff
@@ -1,35 +1,65 @@
 import ActivityCard from "@/components/ActivityCard";
+import React, { useState } from "react";
 import useSWR from "swr";
 import Link from "next/link";
 import styled from "styled-components";
 import PlaceholderLogo from "@/Icons/Placeholder";
 import Navigation from "@/components/Navigation";
 import { theme } from "@/styles";
+import SearchBar from "/components/SearchBar";
 
 export default function HomePage() {
   const { data: activities, isLoading } = useSWR("/api/activities");
-
+  const [filteredActivities, setFilteredActivities] = useState([]);
+  const [searchTerm, setSearchTerm] = useState("");
+  const handleSearch = (searchTerm) => {
+    setSearchTerm(searchTerm);
+    if (!searchTerm) {
+      setFilteredActivities([]);
+      return;
+    }
+    const filtered = activities.filter((activity) =>
+      activity.name.toLowerCase().includes(searchTerm.toLowerCase())
+    );
+    setFilteredActivities(filtered);
+  };
   if (isLoading) return <div>loading...</div>;
   if (!activities) return <div>failed to load</div>;
-
+  let displayedActivities = searchTerm ? filteredActivities : activities;
```

Comment:
const should be work here too.

const displayedActivities = searchTerm ? filteredActivities : activities;